test/e2e_node/cgroup_manager_test.go

```diff
@@ -73,4 +97,283 @@ var _ = framework.KubeDescribe("Kubelet Cgroup Manager [Skip]", func() {
+					podUID        string
+				)
+				By("Creating a Guaranteed pod in Namespace", func() {
+					podName := "qos-pod" + string(uuid.NewUUID())
+					contName := "qos-container" + string(uuid.NewUUID())
+					pod := &api.Pod{
+						ObjectMeta: api.ObjectMeta{
+							Name:      podName,
+							Namespace: f.Namespace.Name,
+						},
+						Spec: api.PodSpec{
+							Containers: []api.Container{
+								{
+									Image:     framework.GetPauseImageName(f.Client),
+									Name:      contName,
+									Resources: getResourceRequirements(getResourceList("100m", "100Mi"), getResourceList("100m", "100Mi")),
+								},
+							},
+						},
+					}
+					podClient := f.PodClient()
+					guaranteedPod = podClient.Create(pod)
+					podUID = string(guaranteedPod.UID)
+				})
+				By("Checking if the pod cgroup was created", func() {
+					podName := "qos-pod" + string(uuid.NewUUID())
+					contName := "qos-container" + string(uuid.NewUUID())
+					podCgroupName := "pod_" + podUID
+					if framework.TestContext.CgroupDriver == "systemd" {
+						podCgroupName = cm.ConvertCgroupNameToSystemd(cm.CgroupName(podCgroupName), true)
+					}
+					fmt.Printf("POD CGROUP NAME CREATE: %v\n", podCgroupName)
+					pod := &api.Pod{
+						ObjectMeta: api.ObjectMeta{
+							Name:      podName,
+							Namespace: f.Namespace.Name,
+						},
+						Spec: api.PodSpec{
+							Containers: []api.Container{
+								{
+									Image:   "gcr.io/google_containers/busybox:1.24",
+									Name:    contName,
+									Command: []string{"sh", "-c", "if [ -d /tmp/memory/" + podCgroupName + " ] && [ -d /tmp/cpu/" + podCgroupName + " ]; then exit 0; else exit 1; fi"},
+									VolumeMounts: []api.VolumeMount{
+										{
+											Name:      "sysfscgroup",
+											MountPath: "/tmp",
+										},
+									},
+								},
+							},
+							Volumes: []api.Volume{
+								{
+									Name: "sysfscgroup",
+									VolumeSource: api.VolumeSource{
+										HostPath: &api.HostPathVolumeSource{Path: "/sys/fs/cgroup"},
+									},
+								},
+							},
+						},
+					}
+					podClient := f.PodClient()
+					podClient.Create(pod)
+					err := framework.WaitForPodSuccessInNamespace(f.Client, podName, contName, f.Namespace.Name)
+					Expect(err).NotTo(HaveOccurred())
+				})
+				By("Checking if the pod cgroup was deleted", func() {
+					// delete the guaranteed pod.
+					podClient := f.PodClient()
+					var gp int64 = 1
+					Expect(podClient.Delete(guaranteedPod.Name, &api.DeleteOptions{GracePeriodSeconds: &gp})).NotTo(HaveOccurred())
+					// Now check that the cgroups get deleted eventually
+					podName := "qos-pod" + string(uuid.NewUUID())
+					contName := "qos-container" + string(uuid.NewUUID())
+					podCgroupName := "pod_" + podUID
+					if framework.TestContext.CgroupDriver == "systemd" {
+						podCgroupName = cm.ConvertCgroupNameToSystemd(cm.CgroupName(podCgroupName), true)
+					}
+					fmt.Printf("POD CGROUP NAME DELETE: %v\n", podCgroupName)
```

Comment:
the other issue with this test structure is its flaky because cleanup of pod cgroups is not tied to the pod deletion.  we should have the pod deletion container check the sysfscgroup tree multiple times and not error right away if it still sees the cgroup